cypress/integration/6-crypto/crypto.spec.ts

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import type { MatrixClient } from "matrix-js-sdk/src/matrix";
+import { SynapseInstance } from "../../plugins/synapsedocker";
+
+function waitForEvent(cli: MatrixClient, event: any, check: () => boolean, callback: () => void) {
+    if (check()) {
+        callback();
+    } else {
+        cli.once(event, () => waitForEvent(cli, event, check, callback));
+    }
+}
+
+describe("Cryptography", () => {
+    beforeEach(() => {
+        cy.startSynapse("default").as('synapse').then(
+            synapse => cy.initTestUser(synapse, "Alice"),
+        );
+    });
+
+    afterEach(() => {
+        cy.get<SynapseInstance>('@synapse').then(synapse => cy.stopSynapse(synapse));
+    });
+
+    it("should receive and decrypt encrypted messages", () => {
+        cy.get<SynapseInstance>('@synapse').then(synapse => cy.getBot(synapse, "Beatrice").as('bot'));
+
+        cy.createRoom({
+            initial_state: [
+                {
+                    type: "m.room.encryption",
+                    state_key: '',
+                    content: {
+                        algorithm: "m.megolm.v1.aes-sha2",
+                    },
+                },
+            ],
+        }).as('roomId');
+
+        cy.get<MatrixClient>('@bot').then(bot =>
+            cy.get<string>('@roomId').then(roomId =>
+                cy.inviteUser(roomId, bot.getUserId())
```

Comment:
Might be worth using something like https://github.com/matrix-org/matrix-react-sdk/pull/8645/files#diff-845f38c4fbfce8fadc06d79d0bb879e309353eeaf6b1424622e4aafcf2cb35b6 to minimise the pyramid nesting

Reply:
This has now been merged so should be available to you once you pull